### ISSUE TITLE
Issue installing Kiva on MacOs

### ISSUE THREAD

**Issue by rdaigle007** [kivy/kivy#8192]
**Software Versions**
* Python: Python 3.11.2
* OS: MacOs Monterey 12.6
* Kivy: ??
* Kivy installation method:

Following steps in https://realpython.com/mobile-app-kivy-python/

instructions for setup

  cd kivy
  python3 -m venv ai-me
  cd ai-me
  source bin/activate 	# need to activate the VE to use it.  How often	needed to do?
  python -m pip install kivy

Build failed...	so tried
  pip install --upgrade pip

Then again
  python -m pip install kivy

Still failed.   Disappointing 1st experience with Kiva.  I'm not super knowledgable with python/pip administration.  So if you give tips without instructions, it may not make sense to me.   If installer doesn't work,  makes me wonder how much general issues I'll have with Kivy.  Just want a simple Mobile UI framework for ios apps.  

**Describe the bug**

Shouldn't have issues installing Kivy.  :).   I searched bugs and other internet source.  Not much luck.

Logs of build is below

The issue tracker is a tool to address **bugs only** (search known bugs: https://git.io/vM1iE).
Please use the #support Discord channel at https://chat.kivy.org/ or Stack Overflow for
support questions, more information at https://git.io/vM1yQ.

**Expected behavior**

**To Reproduce**
A short, *runnable* example that reproduces the issue with *latest kivy master*.

**Code and Logs and screenshots**

The full build log is to large.  So I provided a little of the beginning (tons of deprecation warnings.... only showing some in the beginning... then show more of the errors at the end

```
$ python -m pip install kivy
Collecting kivy
  Using cached Kivy-2.1.0.tar.gz (23.8 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting Kivy-Garden>=0.1.4
  Using cached Kivy_Garden-0.1.5-py3-none-any.whl (4.6 kB)
Collecting docutils
  Using cached docutils-0.19-py3-none-any.whl (570 kB)
Collecting pygments
  Using cached Pygments-2.14.0-py3-none-any.whl (1.1 MB)
Collecting requests
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.1.0-cp311-cp311-macosx_10_9_x86_64.whl (123 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.12.7-py3-none-any.whl (155 kB)
Building wheels for collected packages: kivy
  Building wheel for kivy (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for kivy (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [1096 lines of output]
      [INFO   ] [Logger      ] Record log in /Users/russdaigle/.kivy/logs/kivy_23-04-02_11.txt
      [INFO   ] [Kivy        ] v2.1.0
      [INFO   ] [Kivy        ] Installed at "/private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/__init__.py"
      [INFO   ] [Python      ] v3.11.2 (main, Feb 16 2023, 03:15:23) [Clang 14.0.0 (clang-1400.0.29.202)]
      [INFO   ] [Python      ] Interpreter at "/Users/russdaigle/code/kivy/ai-me/bin/python"
      [INFO   ] [Logger      ] Purge log fired. Processing...
      [INFO   ] [Logger      ] Purge finished!
       <string>:370: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
       <string>:373: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
       <string>:377: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
       warning: kivy/graphics/common.pxi:9:4: 'const_char_ptr' redeclared
       warning: kivy/graphics/common.pxi:23:4: 'size_t' redeclared
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/weakproxy.c:12479:5: error: incomplete definition of type 'struct _frame'
          __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/weakproxy.c:438:62: note: expanded from macro '__Pyx_PyFrame_SetLineNumber'
        #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
                                                            ~~~~~~~^
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/pytypedefs.h:22:16: note: forward declaration of 'struct _frame'
      typedef struct _frame PyFrameObject;
                     ^
      1 error generated.
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/_event.c:18650:21: warning: fallthrough annotation in unreachable code [-Wunreachable-code-fallthrough]
                          CYTHON_FALLTHROUGH;
                          ^
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/_event.c:284:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
            #define CYTHON_FALLTHROUGH __attribute__((fallthrough))
                                       ^
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/_event.c:18661:21: warning: fallthrough annotation in unreachable code [-Wunreachable-code-fallthrough]
                          CYTHON_FALLTHROUGH;
                          ^
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/_event.c:284:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
            #define CYTHON_FALLTHROUGH __attribute__((fallthrough))


```

COuple errors in the middle:

```
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/graphics/vertex.c:4705:43: warning: 'ob_shash' is deprecated [-Wdeprecated-declarations]
                  hash2 = ((PyBytesObject*)s2)->ob_shash;
                                                ^
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/cpython/bytesobject.h:7:5: note: 'ob_shash' has been explicitly marked deprecated here
          Py_DEPRECATED(3.11) Py_hash_t ob_shash;
          ^
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/pyport.h:336:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/graphics/vertex.c:5324:5: error: incomplete definition of type 'struct _frame'
          __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/graphics/vertex.c:438:62: note: expanded from macro '__Pyx_PyFrame_SetLineNumber'
        #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
                                                            ~~~~~~~^
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/pytypedefs.h:22:16: note: forward declaration of 'struct _frame'
      typedef struct _frame PyFrameObject;
                     ^
      2 warnings and 1 error generated.

```

More below.   THrough to the end with errors at the end

```
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/pyport.h:336:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/core/image/img_imageio.c:6305:43: warning: 'ob_shash' is deprecated [-Wdeprecated-declarations]
                  hash2 = ((PyBytesObject*)s2)->ob_shash;
                                                ^
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/cpython/bytesobject.h:7:5: note: 'ob_shash' has been explicitly marked deprecated here
          Py_DEPRECATED(3.11) Py_hash_t ob_shash;
          ^
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/pyport.h:336:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/core/image/img_imageio.c:7542:5: error: incomplete definition of type 'struct _frame'
          __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/core/image/img_imageio.c:438:62: note: expanded from macro '__Pyx_PyFrame_SetLineNumber'
        #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
                                                            ~~~~~~~^
      /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/pytypedefs.h:22:16: note: forward declaration of 'struct _frame'
      typedef struct _frame PyFrameObject;
                     ^
      2 warnings and 1 error generated.
      Current directory is: /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608
      Source and initial build directory is: /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608
      Python path is:
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608
      /Users/russdaigle/code/kivy/ai-me/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-build-env-y4qxl6mn/site
      /usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python311.zip
      /usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11
      /usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-build-env-y4qxl6mn/overlay/lib/python3.11/site-packages
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-build-env-y4qxl6mn/normal/lib/python3.11/site-packages
      /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/modules
      /Users/russdaigle/.kivy/mods


      Found Cython at /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-build-env-y4qxl6mn/overlay/lib/python3.11/site-packages/Cython/__init__.py
      Detected supported Cython version 0.29.28
      User distribution detected, avoid portable command.
      Using this graphics system: OpenGL ES 2
      GStreamer framework not found, fallback on pkg-config
      WARNING: A problem occurred while running pkg-config --libs --cflags gstreamer-1.0 (code 1)

      b"Package gstreamer-1.0 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `gstreamer-1.0.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'gstreamer-1.0' found\n"

      Missing framework /Library/Frameworks/SDL2.framework
      Missing framework /Library/Frameworks/SDL2_ttf.framework
      Missing framework /Library/Frameworks/SDL2_image.framework
      Missing framework /Library/Frameworks/SDL2_mixer.framework
      SDL2 frameworks not found, fallback on pkg-config
      WARNING: A problem occurred while running pkg-config --libs --cflags sdl2 SDL2_ttf SDL2_image SDL2_mixer (code 1)

      b"Package sdl2 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `sdl2.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'sdl2' found\nPackage SDL2_ttf was not found in the pkg-config search path.\nPerhaps you should add the directory containing `SDL2_ttf.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'SDL2_ttf' found\nPackage SDL2_image was not found in the pkg-config search path.\nPerhaps you should add the directory containing `SDL2_image.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'SDL2_image' found\nPackage SDL2_mixer was not found in the pkg-config search path.\nPerhaps you should add the directory containing `SDL2_mixer.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'SDL2_mixer' found\n"

      Xcode detected at b'/Applications/Xcode.app/Contents/Developer', and using OS X12.6 sdk
      WARNING: A problem occurred while running pkg-config --libs --cflags pangoft2 (code 1)

      b"Package pangoft2 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `pangoft2.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'pangoft2' found\n"

      ERROR: Dependency for context.pyx not resolved: config.pxi
      ERROR: Dependency for compiler.pyx not resolved: config.pxi
      ERROR: Dependency for context_instructions.pyx not resolved: config.pxi
      ERROR: Dependency for fbo.pyx not resolved: config.pxi
      ERROR: Dependency for gl_instructions.pyx not resolved: config.pxi
      ERROR: Dependency for instructions.pyx not resolved: config.pxi
      ERROR: Dependency for opengl.pyx not resolved: config.pxi
      ERROR: Dependency for opengl_utils.pyx not resolved: config.pxi
      ERROR: Dependency for shader.pyx not resolved: config.pxi
      ERROR: Dependency for stencil_instructions.pyx not resolved: config.pxi
      ERROR: Dependency for scissor_instructions.pyx not resolved: config.pxi
      ERROR: Dependency for texture.pyx not resolved: config.pxi
      ERROR: Dependency for vbo.pyx not resolved: config.pxi
      ERROR: Dependency for vertex.pyx not resolved: config.pxi
      ERROR: Dependency for vertex_instructions.pyx not resolved: config.pxi
      ERROR: Dependency for cgl.pyx not resolved: config.pxi
      ERROR: Dependency for cgl_mock.pyx not resolved: config.pxi
      ERROR: Dependency for cgl_gl.pyx not resolved: config.pxi
      ERROR: Dependency for cgl_glew.pyx not resolved: config.pxi
      ERROR: Dependency for cgl_sdl2.pyx not resolved: config.pxi
      ERROR: Dependency for svg.pyx not resolved: config.pxi
      Building extensions in parallel using 4 cores
      Updated build directory to: build/lib.macosx-12-x86_64-cpython-311
      Build configuration is:
       * use_rpi = 0
       * use_egl = 0
       * use_opengl_es2 = 1
       * use_opengl_mock = 0
       * use_sdl2 = 0
       * use_pangoft2 = 0
       * use_ios = 0
       * use_android = 0
       * use_mesagl = 0
       * use_x11 = 0
       * use_wayland = 0
       * use_gstreamer = 0
       * use_avfoundation = 1
       * use_osx_frameworks = 1
       * debug_gl = 0
       * kivy_sdl_gl_alpha_size = 8
       * debug = False
      Updated build/lib.macosx-12-x86_64-cpython-311/kivy/include/config.h
      Updated /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/include/config.h
      Updated build/lib.macosx-12-x86_64-cpython-311/kivy/include/config.pxi
      Updated /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/include/config.pxi
      Updated build/lib.macosx-12-x86_64-cpython-311/kivy/setupconfig.py
      Updated /private/var/folders/rz/hkss8mjn4gx5vq6n9511f_m40000gn/T/pip-install-fm8b2v6c/kivy_9d31db2a442648178ed158bffa1fb608/kivy/setupconfig.py
      Detected compiler is unix
       error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for kivy
Failed to build kivy
ERROR: Could not build wheels for kivy, which is required to install pyproject.toml-based projects
```

**Additional context**

**Comment by misl6** [comment]
Hi @rdaigle007 !

That seems a duplicate of https://github.com/kivy/kivy/issues/8042.

I've posted [here](https://github.com/kivy/kivy/issues/8042#issuecomment-1312419599) a workaround, but the good news is that we will have a new release that will support Python **3.11** soon.

**Comment by rdaigle007** [comment]
Much thanks!  Those 2 commands worked... I think.  I'll try some demo code now to verify.
